### ISSUE TITLE
Bug in Profiler

### ISSUE THREAD

**Issue by IQS-Leopold** [OpenCover/opencover#728]
Please provide the following information when submitting an issue. 

> Where appropriate replace the `[ ]` with a `[X]`

### My Framework

- [ ] .NET 2
- [ ] .NET 3.5
- [ ] .NET 4
- [x] .NET 4.5
- [ ] .NET 4.6
- [ ] .NET 4.6.1
- [ ] .NET 4.6.2
- [ ] .NET Core 1.0.0
- [ ] Something else

### My Environment

- [ ] Windows 7 or below (not truly supported due to EOL)
- [ ] Windows 8
- [ ] Windows 8.1
- [x] Windows 10
- [ ] Windows 10 IoT Core
- [ ] Windows Server 2012
- [ ] Windows Server 2012 R2
- [ ] Windows Server 2016

### I have already...

- [x] repeated the problem using the latest stable release of OpenCover.
- [x] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf).
- [x] have looked at the opencover output xml file in an attempt to resolve the issue.
- [ ] reviewed the [current issues](https://github.com/opencover/opencover/issues) to check that the issue isn't already known.

### My issue is related to (check only those which apply):

- [x] no coverage being recorded
- [ ] 32 or 64 bit support
- [ ] feature request

### Expected Behavior

All assemblies loaded in the application domain which arent excluded by a filter should be tracked.

### Actual Behavior

Assemblies won't be tracked if there is a large assembly e.g (5820 classes). This comes from the fact that there is a timeout in the ProfilerCommunication class in the profiler project. The timeout is specified as COM_WAIT_LONG which is 60000 ms. This is not enough for building the instrumentation model. The Method is: TrackMethod. I think the correct value should be the value of the variable: _short_wait of the same class. 

I also think it would be neccessary to allow the specifiaction of the variable by the commandline not only by environment variable.

### Steps to reproduce the problem:

Create a large assembly with more than 5000 complex classes and try to profile that.

I use open Cover for integration testing.

**Comment by kevinbosman** [comment]
I've encountered the same problem trying to cover a legacy assembly with just a handful of very large methods (>5000 lines each)

I also found that the Profiler timeouts are far too short.

The `COM_WAIT_LONG` timeout is hard coded at 60 seconds and the `_short_wait`, while it can be configured via the `-communicationtimeout` command line option, has an artificially imposed maximum limitation also of just 60 seconds.

I found that in my case I needed to increase **BOTH** the `COM_WAIT_LONG` timeout **AND** the `_short_wait` to at least *240 seconds*, otherwise it was silently timing out with either zero coverage or the dreaded `No results, this could be for a number of reasons` failure.

I would also suggest and request:
1. using `_short_wait` in place of `COM_WAIT_LONG` in `TrackAssembly()` here https://github.com/OpenCover/opencover/blob/67980c8132b1d574beb1feaa7d269c9dca47b812/main/OpenCover.Profiler/ProfilerCommunication.cpp#L466
2. removing the artificial 60s upper limit on the `-communicationtimeout` command line option here https://github.com/OpenCover/opencover/blob/67980c8132b1d574beb1feaa7d269c9dca47b812/main/OpenCover.Framework/CommandLineParser.cs#L248 and on `_shortwait` in `OpenCoverInitialise()` here https://github.com/OpenCover/opencover/blob/67980c8132b1d574beb1feaa7d269c9dca47b812/main/OpenCover.Profiler/CodeCoverage.cpp#L135-L136
